collector/mongos/sharding_status.go

```diff
@@ -0,0 +1,245 @@
+	err := session.DB("config").C("locks").Find(bson.M{ "_id" : "balancer" }).One(&balancerLock)
+	if err != nil {
+		glog.Error("Failed to execute find query on 'config.locks'!")
+	}
+	return balancerLock
+}
+
+func IsBalancerEnabled(session *mgo.Session) float64 {
+	balancerConfig := struct {
+		Stopped	bool	`bson:"stopped"`
+	}{}
+	err := session.DB("config").C("settings").Find(bson.M{ "_id" : "balancer" }).One(&balancerConfig)
+	if err != nil {
+		return 1
+	}
+	if balancerConfig.Stopped == true {
+		return 0
+	}
+	return 1
+}
+
+func IsClusterBalanced(session *mgo.Session) float64 {
+	// Different thresholds based on size
+	// http://docs.mongodb.org/manual/core/sharding-internals/#sharding-migration-thresholds
+	var threshold float64 = 8
+	totalChunkCount := GetTotalChunks(session)
+	if totalChunkCount < 20 {
+		threshold = 2
+	} else if totalChunkCount < 80 && totalChunkCount > 21 {
+		threshold = 4
+	}
+
+	var minChunkCount float64 = -1
+	var maxChunkCount float64 = 0
+	shardChunkInfoAll := GetTotalChunksByShard(session)
+	for _, shard := range *shardChunkInfoAll {
+		if shard.Chunks > maxChunkCount {
+			maxChunkCount = shard.Chunks
+		}
+		if minChunkCount == -1 || shard.Chunks < minChunkCount {
+			minChunkCount = shard.Chunks
+		}
+	}
+
+	// return true if the difference between the min and max is < the thresold
+	chunkDifference := maxChunkCount - minChunkCount
+	if chunkDifference < threshold {
+		return 1
+	}
+
+	return 0
+}
+
+func (status *ShardingStats) Export(ch chan<- prometheus.Metric) {
+	if status.Changelog != nil {
+		status.Changelog.Export(ch)
+	}
+	if status.Topology != nil {
+		status.Topology.Export(ch)
+	}
+	if status.Version != nil {
+		clusterId := status.Version.ClusterId.Hex()
+		mongosClusterIdHex.WithLabelValues(clusterId).Set(1)
+		mongosClusterCurrentVersion.Set(status.Version.CurrentVersion)
+		mongosClusterMinCompatVersion.Set(status.Version.MinCompatVersion)
+	}
+	if status.Mongos != nil && status.BalancerLock != nil {
+		mongosBalancerLockWho := strings.Split(status.BalancerLock.Who, ":")
+		mongosBalancerLockHostPort := mongosBalancerLockWho[0] + ":" + mongosBalancerLockWho[1]
+		mongosBalancerLockTimestamp.WithLabelValues(mongosBalancerLockHostPort).Set(float64(status.BalancerLock.When.Unix()))
+		for _, mongos := range *status.Mongos {
+			labels := prometheus.Labels{"name": mongos.Name, "version": mongos.MongoVersion }
+			mongosUpSecs.With(labels).Set(mongos.Up)
+			mongosPing.With(labels).Set(float64(mongos.Ping.Unix()))
+			mongosBalancerLockState.With(labels).Set(-1)
+			if mongos.Name == mongosBalancerLockHostPort {
+				mongosBalancerLockState.With(labels).Set(status.BalancerLock.State)
+			}
+		}
+	}
+	balancerIsEnabled.Set(status.IsBalanced)
+	balancerChunksBalanced.Set(status.BalancerEnabled)
+
+	balancerIsEnabled.Collect(ch)
+	balancerChunksBalanced.Collect(ch)
+	mongosClusterIdHex.Collect(ch)
+	mongosClusterCurrentVersion.Collect(ch)
+	mongosClusterMinCompatVersion.Collect(ch)
```

Comment:
Removing quite a few of those would be breaking changes to the grafana templates..which ones do you mean, or all?